filebeat/beater/filebeat.go

```diff
@@ -48,10 +51,10 @@ func (fb *Filebeat) Run(b *beat.Beat) error {
 	var wgEvents *sync.WaitGroup // count active events for waiting on shutdown
 	var finishedLogger publisher.SuccessLogger
 
-	if fb.config.ShutdownTimeout > 0 {
-		wgEvents = &sync.WaitGroup{}
-		finishedLogger = newFinishedLogger(wgEvents)
-	}
+	//if fb.config.ShutdownTimeout > 0 || *once {
```

Comment:
it's correct. as long as wgEvent.Wait is not installed in sigWait there is no difference in initializing these always. I just didn't want to have all the wgEvent update by default, but overhead should be ok.